src/Microsoft.OData.Client/DeepInsertSaveResult.cs

```diff
@@ -118,6 +120,65 @@ internal void DeepInsertRequest<T>(T resource)
             }
         }
 
+        /// <summary>
+        /// Asynchronous deep insert request.
+        /// </summary>
+        /// <typeparam name="T">The type of the top-level object to be deep inserted.</typeparam>
+        /// <param name="resource">The top-level object of the type to be deep inserted.</param>
+        internal void BeginDeepInsertRequest<T>(T resource)
+        {
+            PerRequest pereq = null;
+
+            if (resource == null)
+            {
+                throw Error.ArgumentNull(nameof(resource));
+            }
+
+            BuildDescriptorGraph(this.ChangedEntries, true, resource);
+
+            try
+            {
+                ODataRequestMessageWrapper deepInsertRequestMessage = this.GenerateDeepInsertRequest();
+                this.Abortable = deepInsertRequestMessage;

```

Comment:
@KenitoInc What does this one do?

Reply:
Sets the request object that is used in `DataServiceContext.CancelRequest()` method;